vignettes/datatable-joins.Rmd

```diff
@@ -0,0 +1,687 @@
+---
+title: "Joins in data.table"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Joins in data.table}
+  \usepackage[utf8]{inputenc}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r, echo = FALSE, message = FALSE}
+require(data.table)
+knitr::opts_chunk$set(
+  comment = "#",
+    error = FALSE,
+     tidy = FALSE,
+    cache = FALSE,
+ collapse = TRUE
+)
+```
+
+In this vignette you will learn how to perform any join operation using next resources available in the `data.table` syntax.
+
+It assumes familiarity with the `data.table` syntax. If that is not the case, please read the following vignettes:
+
+- `vignette("datatable-intro", package="data.table")`
+- `vignette("datatable-reference-semantics", package="data.table")`
+- `vignette("datatable-keys-fast-subset", package="data.table")`
+
+***
+
+## 1. Defining example data
+
+To illustrate how to use the method available with real life examples, let's simulate a **normalized database** from a little supermarket by performing the following steps:
+
+1. Defining a `data.table` where each product is represented by a row with some qualities, but leaving one product without `id` to show how the framework deals with ***missing values***.
+
+```{r}
+Products = data.table(
+  id = c(1:4, 
+         NA_integer_),
+  name = c("banana",
+           "carrots",
+           "popcorn",
+           "soda",
+           "toothpaste"),
+  price = c(0.63,
+            0.89,
+            2.99,
+            1.49,
+            2.99),
+  unit = c("unit",
+           "lb",
+           "unit",
+           "ounce",
+           "unit"),
+  type = c(rep("natural", 2L),
+           rep("processed", 3L))
+)
+
+Products
+```
+
+2. Defining a `data.table` showing the proportion of taxes to be applied for processed products based on their units.
+
+```{r}
+NewTax = data.table(
+  unit = c("unit","ounce"),
+  type = "processed",
+  tax_prop = c(0.65, 0.20)
+)
+
+NewTax
+```
+
+
+3. Defining a `data.table` simulating the products received every Monday with a `product_id` that is not present in the `Products` table.
+
+```{r}
+set.seed(2156)
+
+ProductReceived = data.table(
+  id = 1:10,
+  date = seq(from = as.IDate("2024-01-08"), length.out = 10L, by = "week"), 
+  product_id = sample(c(NA_integer_, 1:3, 6L), size = 10L, replace = TRUE),
+  count = sample(c(50L, 100L, 150L), size = 10L, replace = TRUE)
+)
+
+ProductReceived
+```
+
+4. Defining a `data.table` to show some sales that can take place on weekdays with another `product_id` that is not present in the `Products` table.
+
+```{r}
+sample_date = function(from, to, size, ...){
+  all_days = seq(from = from, to = to, by = "day")
+  weekdays = all_days[wday(all_days) %in% 2:6]
+  days_sample = sample(weekdays, size, ...)
+  days_sample_desc = sort(days_sample)
+  return(days_sample_desc)
+}
+
+set.seed(5415)
+
+ProductSales = data.table(
+  id = 1:10,
```

Comment:
That would be challenging, as I would need to modify the examples below, since they are related. However, I do not expect the user to pay much attention to that detail.

We can consider avoiding the table display.

Reply:
it is ok to keep all ten rows if necessary, but smaller is better in terms of learning examples usually.